python/dlisio/__init__.py

```diff
@@ -379,12 +382,27 @@ def object(self, type, name, origin, copynr):
         MKAP
 
         """
-        fingerprint = core.fingerprint(type, name, origin, copynr)
-        try:
-            return self.indexedobjects[type][fingerprint]
-        except KeyError:
-            msg = "Object {}.{}.{} of type {} is not found"
-            raise ValueError(msg.format(name, origin, copynr, type))
+        if origin is None or copynr is None:
+            obj = list(self.match('^'+name+'$', type))
+            if len(obj) == 1:
+                return obj[0]
+            elif len(obj) == 0:
+                msg = "No objects with name {} and type {} are found"
+                raise ValueError(msg.format(name, type))
+            elif len(obj) > 1:
+                msg = "Found several objects with requested parameters: \n{}"
+                desc = ""
+                for o in obj:
+                    desc += ("Object {} with origin {} and copynumber {}. \n"
+                             .format(o, o.origin, o.copynumber))
```

Comment:
I think this warning have the potential to be very long with the current format. what about: 
There are multiple <_type_>'s named <_name_>. Found: (origin=1, copy=1), (origin=1, copy=2), (origin=2, copy=1)